internal/response_asserter/response_asserter.go

```diff
@@ -72,12 +88,47 @@ func (a ResponseAsserter[ResponseType]) DecodeAndAssertSingleFields(responsePayl
 	return actualResponse, nil
 }
 
-func (a ResponseAsserter[ResponseType]) DecodeAndAssert(responsePayload []byte) (ResponseType, error) {
-	actualResponse, err := a.DecodeAndAssertSingleFields(responsePayload)
+func (a ResponseAsserter[ResponseType]) DecodeAndAssert(response kafka_client.Response) (ResponseType, error) {
+	actualResponse, err := a.DecodeAndAssertSingleFields(response)
 
 	if err != nil {
 		return actualResponse, err
 	}
 
 	return actualResponse, a.Assertion.AssertAcrossFields(actualResponse, a.Logger)
 }
+
+func (a ResponseAsserter[ResponseType]) assertMessageLength(response kafka_client.Response) error {
+	if len(response.RawBytes) < 4 {
+		return fmt.Errorf("Response is not long enough to accomodate message size.")
+	}
+
+	messageSize := int32(binary.BigEndian.Uint32(response.RawBytes[0:4]))
+
+	if messageSize == int32(len(response.Payload)) {
+		return nil
+	}
+
+	a.Logger.Errorln("❌ Invalid response:")
```

Comment:
Don't think I see a fixture for this, can we add one please? Want to see what the final error message looks like - isn't clear to me atm